h2/connection.py

```diff
@@ -678,8 +690,10 @@ def get_next_available_stream_id(self):
 
         next_stream_id = self.highest_outbound_stream_id + 2
         if next_stream_id > self.HIGHEST_ALLOWED_STREAM_ID:
+            self.config.logger.exception("Exhausted allowed stream IDs")
```

Comment:
Don't bother firing logs when we raise exceptions, our users can do that themselves.